tests/test_backend.py

```diff
@@ -578,12 +545,24 @@ def test_versions(tmpdir, file_name, ext, backend):
         file_name,
     )
 
+    # empty backend
     assert not backend.versions(remote_file, ext=ext)
     with pytest.raises(RuntimeError):
         backend.latest_version(remote_file, ext=ext)
+
+    # v1
     backend.put_file(local_file, remote_file, '1.0.0', ext=ext)
     assert backend.versions(remote_file, ext=ext) == ['1.0.0']
     assert backend.latest_version(remote_file, ext=ext) == '1.0.0'
+
+    # v2
     backend.put_file(local_file, remote_file, '2.0.0', ext=ext)
     assert backend.versions(remote_file, ext=ext) == ['1.0.0', '2.0.0']
     assert backend.latest_version(remote_file, ext=ext) == '2.0.0'
+
+    # v3 with a different extension
```

Comment:
This tests that versions of files with same path but a different extension are not returned.

Reply:
yes

Reply:
Interesting, seems we are lucky that this has had no unwanted consequences in other packages.

Reply:
Would you prefer that we fix it in a separate PR?

Reply:
Is it easy to separate from the rest? If not it should be fine to do it here.